mongoimport/options.go

```diff
@@ -55,8 +55,12 @@ type IngestOptions struct {
 	// Forces mongoimport to halt the import operation at the first insert or upsert error.
 	StopOnError bool `long:"stopOnError" description:"stop importing at first insert/upsert error"`
 
-	// Modifies the import process to update existing objects in the database if they match --upsertFields.
-	Upsert bool `long:"upsert" description:"insert or update objects that already exist"`
+	// Modifies the import process,
+	// "upsert": replace existing objects in the database if they match --upsertFields.
+	// "modify": similar as "upsert', but modify existing object rather than replace them
+	Mode string `long:"mode" value-name:"<mode>" default:"insert" description:"upsert: insert or replace existing objects. merge: modify existing objects rather than replace them. defaults to insert"`
```

Comment:
Use `choice:"insert" choice:"upsert" choice:"merge"` to restrict this option to those choices.